src/Language/PureScript/TypeChecker/Kinds.hs

```diff
@@ -444,7 +444,7 @@ unifyKindsWithFailure onFailure = go
       onFailure w1 w2
 
   unifyRows r1 r2 = do
-    let (matches, rest) = alignRowsWith go r1 r2
+    let (matches, rest) = alignRowsWith (\l t1 t2 -> withErrorMessageHint (ErrorInRowLabel l) $ go t1 t2) r1 r2
```

Comment:
This is a nitpick, but maybe we could extract this lambda (and others) to a binding?
```hs
goWithLabel l t1 t2 = withErrorMessageHint (ErrorInRowLabel l) $ go t1 t2 
```

Reply:
Fixed, I don't know if I agree that this is nicer but I trust your judgement. :)